src/middlewared/middlewared/api/base/server/ws_handler/rpc.py

```diff
@@ -0,0 +1,301 @@
+import asyncio
+import binascii
+from collections import defaultdict
+import enum
+import errno
+import pickle
+import sys
+import traceback
+from typing import Any, Callable
+
+from aiohttp.http_websocket import WSCloseCode, WSMessage
+from aiohttp.web import WebSocketResponse, WSMsgType
+import jsonschema
+
+from truenas_api_client import json
+from truenas_api_client.jsonrpc import JSONRPCError
+
+from middlewared.schema import Error
+from middlewared.service_exception import (CallException, CallError, ValidationError, ValidationErrors, adapt_exception,
+                                           get_errname)
+from middlewared.utils.debug import get_frame_details
+from middlewared.utils.lock import SoftHardSemaphore, SoftHardSemaphoreLimit
+from middlewared.utils.origin import Origin
+from .base import BaseWebSocketHandler
+from ..app import App
+from ..method import Method
+
+REQUEST_SCHEMA = {
```

Comment:
This is good. Would it be helpful to define the response schema somewhere?

Reply:
Why? We don't validate it.

Reply:
Less for validation purposes and more for ease of development so we know exactly what is being sent between the client and server. I've actually gone ahead and fleshed out the response schema with `TypedDict`s in https://github.com/truenas/api_client/pull/7

Reply:
So why do we need an unused jsonschema (that will eventually be out of date) here if you have well-defined python types in your PR?

Reply:
Yeah we don't need it, my original comment was made before I made the PR